rust/src/metadata.rs

```diff
@@ -717,11 +747,60 @@ impl Deserialize for TransactionMetadata {
                 }
             }
             Ok(())
-        })().map_err(|e| e.annotate("TransactionMetadata"))?;
+        })().map_err(|e| e.annotate("GeneralTransactionMetadata"))?;
         Ok(Self(table))
     }
 }
 
+impl cbor_event::se::Serialize for TransactionMetadata {
+    fn serialize<'se, W: Write>(&self, serializer: &'se mut Serializer<W>) -> cbor_event::Result<&'se mut Serializer<W>> {
+        match &self.native_scripts() {
```

Comment:
There's a unit test at the bottom to test that the serialization is still fine, and the inner parts of these matches are from the codegen anyway.